frontend/src/features/env/mutations.ts

```diff
@@ -1,28 +1,52 @@
 // TODO #4279: Remove after React rollout is complete
 import { useMutation } from 'react-query'
-import { useNavigate } from 'react-router-dom'
+import { useParams } from 'react-router-dom'
+
+import { PublicFormViewDto, switchEnvFeedbackFormBodyDto } from '~shared/types'
 
 import {
   adminChooseEnvironment,
   publicChooseEnvironment,
 } from '~services/EnvService'
 
-export const useEnvMutations = () => {
-  const navigate = useNavigate()
+import { submitSwitchEnvFormFeedback } from './EnvService'
+
+export const useEnvMutations = (
+  feedbackForm: PublicFormViewDto | undefined,
+) => {
+  const { formId } = useParams()
 
   const publicSwitchEnvMutation = useMutation(() => publicChooseEnvironment(), {
     onSuccess: () => window.location.reload(),
   })
 
   const adminSwitchEnvMutation = useMutation(() => adminChooseEnvironment(), {
     onSuccess: () => {
-      navigate('/#!/forms')
-      window.location.reload()
+      // If on form builder page
+      if (window.location.href.indexOf('admin/form/') > -1) {
+        window.location.assign(`#!/${formId}/admin`)
```

Comment:
Isn't this missing a leading `/` before the `#`?

Reply:
yes thanks for spotting! have updated it